### PR TITLE
Added documentaiton for BBCOREPOINTER

### PR DESCRIPTION
**Pull request by ohm314**

### CODE REVIEW

File: docs/userdoc/MemoryManagement/bbcorepointer.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+User-allocated data can be managed in NMODL using the `POINTER` type. It allows the
+programmer to reference data that has been allocated in HOC or in VERBATIM blocks. This
+allows for more advanced data-structures that are not natively supported in NMODL.
+
+Since NEURON itself has no knowledge of the layout and size of this data it cannot
+transfer `POINTER` data automatically to CoreNEURON. Furtheremore, in many cases there
+is no need to transfer the data between the two instances. In some cases, however, the
+programmer would like to transfer certain user-defined data into CoreNEURON. The most
+prominent example are random123 RNG stream parameters used in synapse mechanisms. To
+support this use-case the `BBCOREPOINTER` type. Variables that are declared as
```
Comment (pramodk) [inline docs/userdoc/MemoryManagement/bbcorepointer.md:14]:
```suggestion
support this use-case the `BBCOREPOINTER` type <missing something here?>. Variables that are declared as
```

Comment (ohm314) [reply docs/userdoc/MemoryManagement/bbcorepointer.md]: oopsie, forgot to finish sentence...

**Comment by nrnhines** [comment]
Nice, Omar. Very helpful!

**Comment by pramodk** [comment]
@ohm314 : the doc entry is missing? I dont see it in http://bluebrain.github.io/CoreNeuron/index.html

**Pull request merged**